components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -625,6 +627,36 @@ private String getSearchFilter(String attributeName, String filterOperation, Str
         return searchFilter;
     }
 
+    /**
+     * Format the attribute value for search based on the filter operation.
+     *
+     * @param filterOperation Filter operation
+     * @param attributeValue  Value of the attribute
+     * @return Formatted attribute value for search
+     */
+    private String formatSearchAttributeValue(String filterOperation, String attributeValue) {
```

Comment:
Did we test this with multiple filters.

`name co and`

How this is processed through the filtering nodes.
As there is no value after and operation, if we can process `and` part as a value.

Reply:
When using a filter like:

`name co and`, 

`and` is treated as part of the search value (not a logical operator).
The function will escape and format `and` as: `'%and%'`. This means the SQL query will match any name column that contains the word `and`. Since no logical operator is present after `and`, it is not treated as part of a condition but rather as a string value inside the search.